tests/infra_test.go

```diff
@@ -975,3 +969,32 @@ func getKeysFromMetrics(metrics map[string]float64) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+func checkError(err error) error {
+	var e *neturl.Error
+	if errors.As(err, &e) && e.Timeout() {
+		return nil
+	} else {
+		return err
+	}
+}
+
+func validatedHTTPResponses(errorsChan chan error, concurrency int) error {
```

Comment:
Please add a doctext here and explain what it validates, it took me some time to understand this after understanding a day before.